### ISSUE TITLE
bug: Ctrl+← detach 兜底依赖终端发送 ctrl+left 转义序列，macOS Terminal.app 等默认失效

### ISSUE THREAD

**Issue by ccccyk0919** [zhuxixi/pi-agent-board#126]
## 现象

attach 进入会话后，按 `Ctrl+←`（issue #89 引入的无条件 detach 兜底）在部分终端不生效——macOS Terminal.app、以及 Windows 上部分终端均无法退出 attach，只能 Ctrl+C/D 强退（连带 shutdown child Pi）。

**实测环境**：macOS + WezTerm 20240203（`enable_kitty_keyboard = true`）即复现，见下方 WezTerm 根因。

## 根因

`Ctrl+←` detach 的实现（`src/ui/pty-attach.ts` 的 `handleInput`）是：

```ts
if (matchesKey(data, Key.ctrl("left"))) {
    this.detach();
    return;
}
```

而 `matchesKey(data, Key.ctrl("left"))`（pi-tui `keys.js` 的 `case "left"` ctrl 分支）只匹配三种序列：

1. `\x1b[1;5D` —— xterm modifyOtherKeys / kitty 方向键带 ctrl 编码（Pi 在 kitty 探测失败时会开启 modifyOtherKeys `\x1b[>4;2m`）；
2. `\x1bOd` —— SS3 带 ctrl；
3. kitty CSI-u 数字键盘方向键（`\x1b[57417;5u` = KP_LEFT）。

**这三者都要求「终端愿意把 Ctrl+Left 作为转义序列发送给应用」。** 但真实终端并不都满足：

- **macOS Terminal.app**：默认对 `Ctrl+←` **不发送任何字节**（苹果终端公认行为，见 apple.stackexchange.com/questions/1626 —— 原文 "In Windows and Linux I could always hit ctrl+← to skip a word... but no dice in the terminal"）。需要手动在 Profiles → Keyboard 添加映射才会发序列。
- **WezTerm（macOS，实测复现）**：即使 `enable_kitty_keyboard = true`，`Ctrl+←/→` 也被 WezTerm 自身的 tab/pane 切换默认快捷键消费（菜单里 `^→` = "Activate the tab to the right"），键事件根本没有转成转义序列发给应用。这是 WezTerm 已知 issue（wezterm#4237 / #3508）：`debug_key_events=true` 时 Ctrl+Up/Down 会打 log，Ctrl+Left/Right 连 log 都没有。
- 其他终端（Warp 默认用 Ctrl+← 做编辑器内按词跳转、部分终端映射为 tab 切换等）也各有差异。

于是 `matchesKey` 永远收不到输入，detach 分支不被触发，`Ctrl+←` 兜底失效。

## 为什么是 issue #89 的测试盲区

#89 的测试（`test-support/detach-gate-smoke.ts` 场景 L/M）只验证了「注入 `\x1b[1;5D` 时能 detach」，从未验证「真实终端会发送 `\x1b[1;5D`」。U1 验收（合并后用户实测）实际上暴露了该失效，但未作为问题记录。

## 复现

1. 在 macOS Terminal.app（或 macOS + WezTerm，`enable_kitty_keyboard = true`）打开 pi，进入 agent-board attach 任意会话；
2. 编辑器内输入任意文字（使 `←` 门禁判定为非空、转发光标左移）；
3. 按 `Ctrl+←` —— 无反应，无法 detach；
4. 只能 Ctrl+C/D 强退，child Pi 被 shutdown。

## 关联

- 这是 #103（`←` detach 间歇失效）的兜底通道本身失效：当 #103 的 `←` 被吞时，用户本应能靠 `Ctrl+←` 逃生，但该通道在 macOS Terminal.app 等终端同样不可用。
- #89 否决了 Esc（vim 冲突）和双击 ←（误触），但未考虑「Ctrl+← 依赖终端发送能力」这一前提。

## 建议修复方向（待讨论）

WezTerm 的实测证据说明：**「方向键 + modifier」作为唯一退出通道不可靠**——连支持 kitty protocol 的现代终端都可能在 macOS 上吞掉 Ctrl+←。而单按 `←`（`\x1b[D`）是所有终端都一定发送、无任何终端会吞的键。因此建议：

1. **主路径回归 `←` 单键**：修好 #103 的门禁判定（「编辑器空 → `←` detach」100% 可靠），并把 header 提示改为「清空输入后按 ← 退出」，把 `←` 作为真正的终端无关退出通道；
2. **Ctrl+← 降级为可选快键**：保留给能正常发送的终端（现代 Linux 终端等）当快捷方式，不再作为唯一兜底；
3. （可选）若仍要一个「无条件强制退出」，需重新评估从 child 手里抢一个 ASCII 控制键（如 Ctrl+D 语义改为退出确认），涉及产品权衡，单独讨论。

**Comment by ccccyk0919** [comment]
## 状态更新（2026-09-20）

**核心问题已通过 #103 修复缓解。** #103 的修复 PR #119（`fix(attach): editor-shaped ← detach anchor + restore the editor-state reporter endpoint`）已合并，单键 `←` 在编辑器为空时能可靠 detach——「用户被困」的核心后果已解决。

但 #126 建议方向里还剩一个**文案级小尾巴**未做（PR #119 的 plan 明确写了「不动 Ctrl+←」）：

### 剩余待办（待 fix）

`src/ui/pty-attach.ts` 里两处 header 文案仍在把失效的 Ctrl+← 当主退出键宣传：

1. `render()` 的 header 行：
   ```ts
   this.theme.fg("muted", `${this.status} · click opens links · dblclick/drag selects+copies · ←/Ctrl+← detach`);
   ```
2. `renderLoading()` 的中心提示行：
   ```ts
   out.push(center(this.theme.fg("dim", "←/Ctrl+← to detach"), width));
   ```

### 建议的改法

- 两处文案改成「清空输入后按 ← 退出」语义，去掉对 Ctrl+← 的并列宣传；
- `handleInput` 里的 `Key.ctrl("left")` 分支保留为**可选快键**即可（现代终端里能用，但不再承担「唯一兜底」的宣传责任），可顺带把 #89 那段注释措辞从「unconditional detach chord」调整为「可选快键，非主退出通道」。

### 说明

- Ctrl+← 在多终端失效本身是**终端层限制**（WezTerm macOS 菜单栏快捷键拦截，见 wezterm#4237/#3508；Terminal.app 默认不发），代码无法让它在这些终端生效，只能降级 + 靠 `←` 主通道兜底。
- 本 issue 保持 OPEN，待上述 fix 后关闭。

**Comment by zhuxixi** [comment]
开始处理（github-issue-driven 流程）。

D1 之后的形势更新：issue 建议的「主路径回归 ← 单键」已被 #135 架构性完成——`←` 在 `editorEmpty === true`（结构化侧信道）时已 100% 可靠，不再依赖 buffer 启发式。剩余缺口收窄为：**坏终端（macOS Terminal.app / WezTerm-macOS 等）+ `editorEmpty` 为 false/null 时，无任何可用的无条件逃生键**。

接下来：多轮调研（逃生通道设计空间的终端可达性验证 + #89/#48/#103 历史决策考古）→ spec（含验收矩阵）→ 用户确认设计 → 实现。

**Comment by zhuxixi** [comment]
## 调研：无条件逃生通道设计空间（2 轮）

**轮 1（历史决策考古）**：#89 否决 Esc（vim 冲突）与双击 ←（误触），选 Ctrl+← 因组合键意图明确——但只测了「注入序列能 detach」未验证「终端会发序列」（本 issue 已指出的盲区）。#42 裁决 ctrl+] 透传；#48 确立永可退出红线。D1（PR #135）后 `←` 在 editorEmpty=true 时已 100% 可靠（读结构化状态），缺口收窄为「坏终端 + editorEmpty=false/null」。

**轮 2（候选通道技术验证）**，推荐 **Ctrl+\（0x1c 裸字节）** 作为无条件逃生键：

| 候选 | 终端可达性 | 验证 |
|---|---|---|
| **Ctrl+\（0x1c）** | 最强：ASCII 控制字符，raw mode 下所有终端原样发送（kitty legacy ctrl-mapping 官方表）；无终端默认占它做快捷键 | 实测 `matchesKey('\x1c','ctrl+\\')`===true（pi-tui 原生支持）；pi dist 全量 rg 无 ctrl+\ 绑定；本仓无占用 |
| F 键 | 强但有零星终端绑定冲突 | — |
| 鼠标 detach 区 | 强 | 误触+可发现性差，不单独承担 |
| Esc×3 / 双击 ← | 弱 | #89 已否决 |
| 只修文案 | — | 止血不根治 |

已知代价：从 child 抢 Ctrl+\（Pi 无绑定；仅影响 child 内 vim terminal-mode 的 Ctrl+\ Ctrl+N 极小众路径），文档化即可。

形成三层逃生梯队：`←`（editorEmpty=true，智能）→ `Ctrl+←`（好终端快捷）→ **`Ctrl+\`（无条件，终端无关）**。下一步出 spec（含验收矩阵）。

**Comment by zhuxixi** [comment]
## 已修复（PR #137，CR PASS + CI 双 Node 绿）✅

新增 **`Ctrl+\`（裸字节 0x1c）作为无条件、终端无关的 detach 键**，三层逃生梯队成型：

1. `←` — 智能层：`editorEmpty === true` 时 detach（D1 契约，PR #135）
2. `Ctrl+←` — 快捷层：好终端保留
3. **`Ctrl+\`** — 无条件层：任何编辑器状态、任何连接状态立即 detach，**零终端修饰键编码依赖**（macOS Terminal.app 不发 Ctrl+←、WezTerm-macOS 消费它为 tab 切换——本 issue 复现环境的直接根治）

### 关键验证

- `matchesKey` 对三种真实键盘模式编码全部原生匹配（spec review 实测）：裸 `\x1c`、kitty CSI-u `\x1b[92;5u`、modifyOtherKeys `\x1b[27;5;92~`
- pi 本体 dist 无任何 `ctrl+\` 绑定（全量 rg）
- A1-A4 自动化验收全绿：四象限无条件 detach（editorEmpty=true/false/null × connected + disconnected）、草稿态显式 pin、编码形态 + `q` 转发隔离、header 文案断言；全量 1022/1017（2 fail 均为账本 load flake，隔离复跑干净）；typecheck 干净
- header 附带修了一个 pre-existing bug：原文案 83 列在 80 列终端已截断，现 75 列两个 chord 完全可见

### 已知代价（README 两处已如实登记）

- AltGr 布局（如德式）`\` 在 AltGr 层，可能产不出 0x1c——该场景用 `Ctrl+←`（Linux 现代终端可靠）或 `←`（editorEmpty=true 时）
- legacy 模式下 `Ctrl+4` 产同字节，也会 detach
- attach 期间从 child 抢 `Ctrl+\`（child 内 vim terminal-mode 的 Ctrl+\ 前缀不可用；Pi 无此绑定）

### 验收对账

- A1/A2/A3/A4（自动化）：全部 CLOSED
- U1（macOS Terminal.app 实测）：**pending**（需 mac 环境；步骤：attach 会话 + 编辑器含草稿 → Ctrl+\ → detach 回 dashboard，child 存活）
- U1b（Linux 等效：WezTerm kitty + xterm legacy 各一次草稿态 Ctrl+\ 退出、重新 attach 草稿还在）：待用户执行

后续：#128（attach 吞终端查询 OSC 11/DA）是同会话下的下一个架构级弊端候选。
